### PR TITLE
Gradle 6-driven updates

### PR DESCRIPTION
**Pull request by ejona86**
Each commit will be merged separately.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -492,14 +492,9 @@ subprojects {
...
 
         plugins.withId("java") {
-            task javadocJar(type: Jar) {
-                classifier = 'javadoc'
-                from javadoc
-            }
-
-            task sourcesJar(type: Jar) {
-                classifier = 'sources'
-                from sourceSets.main.allSource
```
Comment (dapengzhang0) [inline build.gradle:502]: The built-in will also pick source from the main source set only?

File: testing-proto/build.gradle [hunk 2]
```diff
@@ -12,7 +12,6 @@ dependencies {
             project(':grpc-stub')
     compileOnly libraries.javax_annotation
     testImplementation libraries.truth
-    testRuntime libraries.javax_annotation
```
Comment (dapengzhang0) [inline testing-proto/build.gradle:15]: Why is this removed. Does it work for all JDK versions?

File: build.gradle (hunk 1 above)
Comment (dapengzhang0) [reply build.gradle:502]: I mean the proto codegen will not be picked.

Comment (ejona86) [reply build.gradle:502]: Yes, it produces the same results. Honestly, it would be better to include the generated code in my mind, but it's not that big of a deal.

File: testing-proto/build.gradle (hunk 2 above)
Comment (ejona86) [reply testing-proto/build.gradle:15]: `testRuntime` is the old-style. It seemed weird to be runtime-only. It seemed like it should be `testImplementation`. So I looked at the code and couldn't find a reference to it, so I deleted it. The Java 11 build did fail, so I put it back, but as `testRuntimeOnly`.
